cmd/kube-proxy/app/server_others.go

```diff
@@ -164,6 +164,56 @@ func newProxyServer(
 			healthzServer,
 			config.NodePortAddresses,
 		)
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
```

Comment:
The check should be 


0. if 1 cidr and `dualstack off` => ok, base case 
1. if 1 cidr and `dualstack on` => ok, create single stack proxy
2. if 2 cidrs and `dualstack on` => ok, create dualstack metaproxy
3. if 2 cidrs set and `dualstack off` => error

The above might fail if the user turned on the feature and set two cidrs. which we allow.

Reply:
ipvs follows this same logic.  If dual-stack is enabled, make a dual-stack proxier.  We want to diminish the meaning of cluster-cidr here, so I do not think testing that should be used.  TL;DR is that we have no real way to know whether dual-stack is enabled or not except to try it, and even that could change at run time.  Very little cost (I think) to setting up both modes, but we do need to handle failure gracefully if dual-stack is enabled on a single-stack node.

Reply:
Thought more about this, I agree with Tim.